### PR TITLE
probes moved to middleware and improved

### PR DESCRIPTION
**Pull request by mjholder**
The probes now exist in a middleware. I have this because requests go through middlewares before views. If something is wrong and we have our probes in the view, the error will propagate before we can check anything ourselves. With the probes being in a middleware, we can run our tests and return results first.

liveness (/livez): we check if we have a valid connection to the db and make sure the cache isn't failing.
readiness (/readyz): we check if there are migrations that need to be run (room for improvement)

ticket: RHCLOUD-27155

### CODE REVIEW

**Review by Victoremepunto — CHANGES_REQUESTED** [review]
Please, can you ...
- Remove the healthz endpoint 
- Add `livenessProbe` and `readinessProbe` definitions in the Clowdapp template to use the two corresponding endpoints (`/livez` and `/readyz`) ? 

Today the Clowdapp defaults to use `/healthz` which is a dummy implementation in our Starter App anyways.

**Comment by mjholder** [comment]
> Please, can you ...
> 
>     * Remove the healthz endpoint
> 
>     * Add `livenessProbe` and `readinessProbe` definitions in the Clowdapp template to use the two corresponding endpoints (`/livez` and `/readyz`) ?
> 
> 
> Today the Clowdapp defaults to use `/healthz` which is a dummy implementation in our Starter App anyways.

Ok I made those changes. Is there a way to test them?

**Comment by Victoremepunto** [comment]
> Ok I made those changes. Is there a way to test them?

Yes, but only on a deployment to a K8s cluster. I think we should add a pr_check Jenkinsfile for this. Gonna create a card. 

You can do this "manually" by running  `make build-image  push-image deploy`

File: {{cookiecutter.project_name}}/clowdapp.yaml [hunk 1]
```diff
@@ -46,6 +56,9 @@ parameters:
 - name: APP_NAME
   required: true
   value: {{cookiecutter.project_name | replace("_", "-")}}
+- name: WEB_PORT
+  description: Port this is running on
+  value: 8000
```
Comment (Victoremepunto) [inline {{cookiecutter.project_name}}/clowdapp.yaml]: Needs quotes

```suggestion
  value: "8000"
```

File: {{cookiecutter.project_name}}/clowdapp.yaml [hunk 2]
```diff
@@ -30,6 +30,16 @@ objects:
...
+        livenessProbe:
+          httpGet:
+            path: /livez
+            port: ${WEB_PORT}
+          initialDelaySeconds: 3
+          periodSeconds: 3
+        readinessProbe:
+          httpGet:
+            path: /readyz
+            port: ${WEB_PORT}
```
Comment (Victoremepunto) [inline {{cookiecutter.project_name}}/clowdapp.yaml]: Use the `${{VALUE}}` syntax for setting as a JSON value (not a String).

See https://docs.openshift.com/container-platform/4.13/openshift_images/using-templates.html#templates-writing-parameters_using-templates

```suggestion
            port: ${{WEB_PORT}}
          initialDelaySeconds: 3
          periodSeconds: 3
        readinessProbe:
          httpGet:
            path: /readyz
            port: ${{WEB_PORT}}
```

**Review by Victoremepunto — CHANGES_REQUESTED** [review]
Almost there, a couple of syntax issues I've spotted on the template.

**Review by Victoremepunto — APPROVED** [review]

**Pull request merged**